corehq/apps/repeaters/models.py

```diff
@@ -287,6 +298,37 @@ def get_auth(self):
             return HTTPDigestAuth(self.username, self.password)
         return None
 
+    def fire_for_record(self, repeat_record):
+        headers = self.get_headers(repeat_record)
+        auth = self.get_auth()
+        payload = self.get_payload_and_handle_exception(repeat_record)
+        url = self.get_url(repeat_record)
+        try:
+            response = simple_post(payload, url, headers=headers, timeout=POST_TIMEOUT, auth=auth)
+        except (Timeout, ConnectionError) as error:
+            log_repeater_timeout_in_datadog(self.domain)
+            self.handle_response(RequestConnectionError(error), repeat_record)
```

Comment:
@proteusvacuum What do you think about this to address your (correct) comment?

The alternative is nesting another try/except in the try block. I'm leaning towards this because it keeps it flat and has the same effect, the only downside being the somewhat surprising choice to pass a newly created exception directly to a function instead of raising it.